Fix flaky docker celery tests

Celery docker tests randomly fail with the following output:


```
=================================== FAILURES ===================================
_________________________ test_custom_tracer_provider __________________________

celery_app = <Celery celery.tests at 0x7f88b2d3cbe0>
memory_exporter = <opentelemetry.sdk.trace.export.in_memory_span_exporter.InMemorySpanExporter object at 0x7f88b8305790>

    def test_custom_tracer_provider(celery_app, memory_exporter):
        @celery_app.task
        def fn_task():
            return 42
    
        resource = resources.Resource.create({})
        tracer_provider = TracerProvider(resource=resource)
        span_processor = export.SimpleSpanProcessor(memory_exporter)
        tracer_provider.add_span_processor(span_processor)
    
        trace_api.set_tracer_provider(tracer_provider)
    
        CeleryInstrumentor().uninstrument()
        CeleryInstrumentor().instrument(tracer_provider=tracer_provider)
    
        fn_task.delay()
    
        spans_list = memory_exporter.get_finished_spans()
>       assert len(spans_list) == 1
E       assert 2 == 1
E         +2
E         -1

celery/test_celery_functional.py:579: AssertionError
```

We need to investigate this and fix th test so it always passes.

--- a/tests/opentelemetry-docker-tests/tests/celery/test_celery_functional.py
+++ b/tests/opentelemetry-docker-tests/tests/celery/test_celery_functional.py
@@ -15,7 +15,6 @@
 
 import celery
 from celery.exceptions import Retry
-from flaky import flaky
 from pytest import mark
 
 import opentelemetry.instrumentation.celery
@@ -559,8 +558,6 @@ def test_apply_async_previous_style_tasks(
     ) == async_run_span.attributes.get(SpanAttributes.MESSAGING_MESSAGE_ID)
 
 
-# FIXME find a permanent solution for the flakiness of this test
-@flaky
 def test_custom_tracer_provider(celery_app, memory_exporter):
     @celery_app.task
     def fn_task():
@@ -576,7 +573,7 @@ def test_custom_tracer_provider(celery_app, memory_exporter):
     CeleryInstrumentor().uninstrument()
     CeleryInstrumentor().instrument(tracer_provider=tracer_provider)
 
-    fn_task.delay()
+    fn_task.apply()
 
     spans_list = memory_exporter.get_finished_spans()
     assert len(spans_list) == 1

